wandb/sdk/wandb_verify.py

```diff
@@ -0,0 +1,495 @@
+                "Server failed to accept a graphql put request with response {}. Check bucket permissions.".format(
+                    status_code
+                ),
+                False,
+            )
+
+            # next test will also fail if this one failed. So terminate this test here.
+            return None
+
+    # wait for upload to finish before download
+    time.sleep(2)
+
+    public_api = wandb.Api()
+    try:
+        prev_run = public_api.run("{}/{}/{}".format(entity, PROJECT_NAME, run_id))
+    except Exception:
+        failed_test_strings.append(
+            "Unable to access previous run through public API. Contact W&B for support."
+        )
+        print_results(failed_test_strings, False)
+        return None
+    try:
+        read_file = prev_run.file(gql_fp).download(replace=True)
+    except Exception:
+        failed_test_strings.append(
+            "Unable to read file successfully saved through a put request. Check SQS configurations, topic configs and SNS configs."
+        )
+        print_results(failed_test_strings, False)
+        return None
+    contents = read_file.read()
+    try:
+        assert contents == "test2"
+    except AssertionError:
+        failed_test_strings.append(
+            "Read file contents do not match saved file contents. Contact W&B for support."
+        )
+
+    print_results(failed_test_strings, False)
+    return url
+
+
+def check_large_file(api: Api, host: str) -> None:
+    print(
+        "Checking ability to send large payloads through proxy".ljust(72, "."), end=""
+    )
+    descy = "a" * int(10 ** 7)
+
+    username = getpass.getuser()
+    failed_test_strings = []
+    query = gql(
+        """
+        query Project($entity: String!, $name: String!, $runName: String!, $desc: String!){
+            project(entityName: $entity, name: $name) {
+                run(name: $runName, desc: $desc) {
+                    name
+                    summaryMetrics
+                }
+            }
+        }
+        """
+    )
+    public_api = wandb.Api()
+    client = public_api._base_client
+
+    try:
+        client._get_result(
+            query,
+            variable_values={
+                "entity": username,
+                "name": PROJECT_NAME,
+                "runName": "",
+                "desc": descy,
+            },
+            timeout=60,
+        )
+    except Exception as e:
+        if isinstance(e, requests.HTTPError) and e.response.status_code == 413:
+            failed_test_strings.append(
+                'Failed to send a large payload. Checl nginx.ingress.kubernetes.io/proxy-body-size is "0".'
+            )
+        else:
+            failed_test_strings.append(
+                "Failed to send a large payload with error: {}.".format(e)
+            )
+    print_results(failed_test_strings, False)
+
+
+def check_wandb_version(api: Api) -> None:
+    print("Checking wandb package version is up to date".ljust(72, "."), end="")
```

Comment:
I think @raubitsj might know a good library to use for portable formatted prints.

Reply:
from future import print should make this work

Reply:
Thanks Jeff!